File: dev/test_data/typescript/test_main/aas_core_meta.v3/input/snippets/Verification/properties_or_ranges_have_value_type.ts
/**
 * Check that `elements` which are {@link types.Property} or {@link types.Range}
 * have the given `valueType`.
 *
 * @param elements - to be verified
 * @returns `true` if the check passes
 */
export function propertiesOrRangesHaveValueType(
  elements: Iterable<AasTypes.ISubmodelElement>,
  valueType: AasTypes.DataTypeDefXsd
): boolean {
  for (const element of elements) {
    if (AasTypes.isProperty(element) || AasTypes.isRange(element)) {
      if (element.valueType !== valueType) {
        return false;
      }
    }
  }

  return true;
}

File: dev/test_data/test_main/python/aas_core_meta.v3/input/snippets/Verification/properties_or_ranges_have_value_type.py
def properties_or_ranges_have_value_type(
    elements: Iterable[aas_types.SubmodelElement], value_type: aas_types.DataTypeDefXSD
) -> bool:
    """
    Check that :paramref:`elements` which are
    :py:class:`.types.Property` or :py:class:`.types.Range`
    have the given :paramref:`value_type`.
    """
    range_or_property = (aas_types.Property, aas_types.Range)
    for element in elements:
        if isinstance(element, range_or_property):
            if element.value_type is not value_type:
                return False

    return True
